pkg/services/ngalert/schedule/schedule.go

```diff
@@ -307,6 +311,26 @@ func (sch *schedule) DroppedAlertmanagersFor(orgID int64) []*url.URL {
 	return s.DroppedAlertmanagers()
 }
 
+// UpdateAlertRule looks for the active rule evaluation and commands it to update the rule
+func (sch *schedule) UpdateAlertRule(key models.AlertRuleKey) {
+	ruleInfo, err := sch.registry.get(key)
+	if err != nil {
+		return
+	}
+	ruleInfo.update()
+}
+
+// DeleteAlertRule stops evaluation of the rule, deletes it from active rules, and cleans up state cache.
+func (sch *schedule) DeleteAlertRule(key models.AlertRuleKey) {
+	ruleInfo, ok := sch.registry.del(key)
+	if !ok {
+		sch.log.Info("unable to get alert rule routine information by key", "uid", key.UID, "org_id", key.OrgID)
```

Comment:
```suggestion
		sch.log.Info("unable to delete alert rule routine information by key", "uid", key.UID, "org_id", key.OrgID)
```

Also very nitpicky, but in the previous wording I suspect someone debugging this might confuse this log message for being associated with a read rather than a delete.

Also, the old log specifically assumed the only way the delete could fail would be that the record didn't exist. If this is true, it may be worth adjusting the statement to mention this, e.g. `Unable to delete rule routine information because it did not exist.`.At first glance I wondered why this log wasn't at error level based on the message, but it was more clear after reading the original context.